app/controllers/course/admin/lesson_plan_settings_controller.rb

```diff
@@ -4,25 +4,50 @@ class Course::Admin::LessonPlanSettingsController < Course::Admin::Controller
   add_breadcrumb :edit, :course_admin_lesson_plan_path
 
   def edit
-    @page_data = @item_settings.lesson_plan_item_settings.to_json
+    @page_data = page_data.to_json
   end
 
   def update
-    if @item_settings.update(lesson_plan_item_settings_params['lesson_plan_item_settings']) &&
+    if update_lesson_plan_items_settings &&
+       update_lesson_plan_component_settings &&
        current_course.save
-      render json: @item_settings.lesson_plan_item_settings
+      render json: page_data
     else
       head :bad_request
     end
   end
 
   private
 
+  def update_lesson_plan_items_settings
+    item_settings_params = lesson_plan_item_settings_params[:lesson_plan_item_settings]
+    item_settings_params.nil? || @item_settings.update(item_settings_params)
+  end
+
+  def update_lesson_plan_component_settings
+    component_settings_params = lesson_plan_item_settings_params[:lesson_plan_component_settings]
+    component_settings_params.nil? || @settings.update(component_settings_params)
+  end
+
   def lesson_plan_item_settings_params
-    params.require(:lesson_plan_settings).permit(lesson_plan_item_settings: {})
+    params.require(:lesson_plan_settings).permit(
+      lesson_plan_item_settings: {},
+      lesson_plan_component_settings: [:milestones_expanded]
+    )
   end
 
   def load_item_settings
     @item_settings = Course::Settings::LessonPlanItems.new(current_component_host.components)
   end
+
+  def page_data
+    {
+      items_settings: @item_settings.lesson_plan_item_settings,
```

Comment:
Using `page_data` seems ok, only difference between this and jbuilder is that jbuilder can be more explicit. Following that line of thought, perhaps we can consider having some tests to test the shape of the output, otherwise changes to settings gem might go unnoticed. Maybe we can test it at the settings itself, rather than at the controller. 

Thoughts?

Reply:
@fonglh has tests for the models involved. They are in `spec/models/course/settings/*`. Do you think we need any more tests?

Reply:
That works 👍